clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -557,6 +558,8 @@ private void addToCompletedReceives() {
         public final Sensor bytesReceived;
         public final Sensor selectTime;
         public final Sensor ioTime;
+        private final List<MetricName> topLevelMetricNames = new ArrayList<>();
```

Comment:
Perhaps add a comment that this is for metrics that are not registered through sensors.